File: src/testharness/session-config-test-builder.ts
import type { AgentSessionConfig } from "../domain/value-objects/agent-session-config.js";

export class SessionConfigTestBuilder {
  private provider = "ollama";
  private modelId = "gemma4";
  private systemPrompt = "You are a helpful assistant.";
  private tools: readonly string[] = [];
  private thinkingLevel = "off";

  private constructor() {}

  static create(): SessionConfigTestBuilder {
    return new SessionConfigTestBuilder();
  }

  withProvider(provider: string): SessionConfigTestBuilder {
    this.provider = provider;
    return this;
  }

  withModelId(modelId: string): SessionConfigTestBuilder {
    this.modelId = modelId;
    return this;
  }

  withSystemPrompt(systemPrompt: string): SessionConfigTestBuilder {
    this.systemPrompt = systemPrompt;
    return this;
  }

  withTools(tools: readonly string[]): SessionConfigTestBuilder {
    this.tools = tools;
    return this;
  }

  withThinkingLevel(thinkingLevel: string): SessionConfigTestBuilder {
    this.thinkingLevel = thinkingLevel;
    return this;
  }

  build(): AgentSessionConfig {
    return {
      provider: this.provider,
      modelId: this.modelId,
      systemPrompt: this.systemPrompt,
      tools: this.tools,
      thinkingLevel: this.thinkingLevel,
    };
  }
}

File: src_csharp/haas/HaaS.Domain.Tests/Builders/SessionConfigTestBuilder.cs
using HaaS.Domain.ValueObjects;

namespace HaaS.Domain.Tests.Builders;

public class SessionConfigTestBuilder
{
    private string _provider = "ollama";
    private string _modelId = "gemma4";
    private string _systemPrompt = "You are a helpful assistant.";
    private IReadOnlyList<string> _tools = [];
    private string _thinkingLevel = "off";

    private SessionConfigTestBuilder() { }

    public static SessionConfigTestBuilder Create() => new();

    public SessionConfigTestBuilder WithProvider(string provider)
    {
        _provider = provider;
        return this;
    }

    public SessionConfigTestBuilder WithModelId(string modelId)
    {
        _modelId = modelId;
        return this;
    }

    public SessionConfigTestBuilder WithSystemPrompt(string prompt)
    {
        _systemPrompt = prompt;
        return this;
    }

    public SessionConfigTestBuilder WithTools(IReadOnlyList<string> tools)
    {
        _tools = tools;
        return this;
    }

    public SessionConfigTestBuilder WithThinkingLevel(string level)
    {
        _thinkingLevel = level;
        return this;
    }

    public AgentSessionConfig Build() => new(_provider, _modelId, _systemPrompt, _tools, _thinkingLevel);
}
